### PR TITLE
CB-4208. Store common salt scripts in common module & handle compression at on…

### PR DESCRIPTION
**Pull request by oleewere**
adding new util file to handle compressing of salt resource.
I preferred to use a static util class instead of a spring bean in order to not create beans for every services that uses common module.

reason fro the change:
- fluentd or databus will be used in freeipa soon. relatively complicated scripts needs to be duplicated there, to avoid that, use the same salt scripts for distrox/sdx and freeipa
(also the definition will differ a bit...only in one file, so it's required to merge common scripts with specified ones...btw, freeipa changes are on freeipa-logging branch) -> the change itself is useful addition without any freeipa related changes, so that can be an individual commit.

changes:
- move compress code to a new class in common module
- move common salt scripts to common module
- make it possible to merge files from multiple folders (so if salt-common and a specified folder contains the same files, that specified will be used )
(note: fluentd & databus salt files will be there for freeipa after that, but it won't have any impact yet as those are not included in top.sls)

tested with env creation (im expecting it should work the same way for sdx/distrox)

### CODE REVIEW

File: common/src/main/java/com/sequenceiq/cloudbreak/util/CompressUtil.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            String folder) throws IOException {
+        ResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+        for (Resource resource : resolver.getResources(String.format("classpath*:%s/**", folder))) {
+            String path = resource.getURL().getPath();
+            String dir = path.substring(path.indexOf(String.format("/%s", folder)) + String.format("/%s", folder).length(), path.lastIndexOf('/') + 1);
+            List<Resource> list = structure.get(dir);
+            if (list == null) {
+                list = new ArrayList<>();
+            }
+            structure.put(dir, list);
```
Comment (lajosrodek) [inline common/src/main/java/com/sequenceiq/cloudbreak/util/CompressUtil.java]: This statement should be inside the `if` block above. But, actually, I suggest replacing lines 59--63 with something like `structure.computeIfAbsent(dir, () -> new ArrayList<>())` that is more compact. 😃

File: common/src/main/java/com/sequenceiq/cloudbreak/util/CompressUtil.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            return baos.toByteArray();
+        }
+    }
+
+    private static void fillStructureWithResources(Map<String, List<Resource>> structure,
+            String folder) throws IOException {
+        ResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+        for (Resource resource : resolver.getResources(String.format("classpath*:%s/**", folder))) {
+            String path = resource.getURL().getPath();
+            String dir = path.substring(path.indexOf(String.format("/%s", folder)) + String.format("/%s", folder).length(), path.lastIndexOf('/') + 1);
```
[the remark this answers is not in the document]
Comment (oleewere) [reply common/src/main/java/com/sequenceiq/cloudbreak/util/CompressUtil.java]: note that change mainly a code-move only, but i can do some small refactor

**Comment by lacikaaa** [comment]
I don't really get why would you move those things to common module. orchestrator-salt is a dependency for freeipa module as well for cb core. So I think it should stay in it's original module

**Comment by oleewere** [comment]
on freeipa node:
```
[root@ipaserver0 cloudbreak]# cd /srv/pillar/
[root@ipaserver0 pillar]# ll
total 4
drw-r--r-- 2 root root 22 Oct 30 16:01 databus
drw-r--r-- 2 root root 22 Oct 30 16:01 fluent
drw-r--r-- 2 root root 22 Oct 30 16:01 freeipa
drwxr--r-- 2 root root 23 Oct 30 16:01 nodes
-rw-r--r-- 1 root root 26 Oct 30 16:01 top.sls
[root@ipaserver0 pillar]# cd ..
[root@ipaserver0 srv]# ll salt/
databus/ dns/     fluent/  freeipa/ nginx/   top.sls
[root@ipaserver0 srv]# ll salt/
total 4
drw-r--r-- 3 root root 42 Oct 30 16:01 databus
drw-r--r-- 4 root root 49 Oct 30 16:01 dns
drw-r--r-- 3 root root 80 Oct 30 16:01 fluent
drw-r--r-- 3 root root 37 Oct 30 16:01 freeipa
drw-r--r-- 3 root root 34 Oct 30 16:01 nginx
-rw-r--r-- 1 root root 48 Oct 30 16:01 top.sls
[root@ipaserver0 srv]#
```

**Pull request merged**